htdocs/google/admin/google_contactsync.php

```diff
@@ -443,15 +464,26 @@
 			$i++;
 		}
 
-		$result=0;
-		if (count($gContacts)) $result=insertGContactsEntries($gdata, $gContacts, $objectstatic);
+		$resultEntries=0;
+		if (count($gContacts)) $resultEntries=insertGContactsEntries($gdata, $gContacts, $objectstatic);
 
-		if (is_numeric($result) && $result >= 0) {
+		if (is_numeric($resultEntries) && $resultEntries >= 0) {
 			$mesg = $langs->trans("PushToGoogleSucess", count($gContacts));
 		} else {
 			$error++;
-			$errors[] = $langs->trans("Error").' '.$result;
+			$errors[] = $langs->trans("Error").' '.$resultEntries;
 		}
+
+		$resultTags=0;
+		if (count($gContacts)) $resultTags=updateGContactGroups($gdata, $gContacts, 'member');
+
+		if (is_numeric($resultTags) && $resultTags >= 0) {
+			$mesg .= '<br>'.$langs->trans("TagsCreatedSuccess");
+		} else {
+			$error++;
+			$errors[] = $langs->trans("Error").' '.$resultTags;
+		}
```

Comment:
Here you update the contact group even if the contact as not been created. If an error occure the database and the google agenda musn't change from it's initial state.

Reply:
https://github.com/DoliCloud/DoliMods/pull/5#discussion_r932097684